File: sample/src/main/java/net/yslibrary/historian/sample/DebugTree.java
package net.yslibrary.historian.sample;

import timber.log.Timber;

class DebugTree extends Timber.DebugTree {
  @Override
  protected String createStackElementTag(StackTraceElement element) {
    return String.format("(%s:%s)#%s",
        element.getFileName(),
        element.getLineNumber(),
        element.getMethodName());
  }
}

File: sample/src/main/java/net/yslibrary/historian/sample/DebugTree.kt
package net.yslibrary.historian.sample

import timber.log.Timber

internal class DebugTree : Timber.DebugTree() {
    override fun createStackElementTag(element: StackTraceElement): String {
        return String.format("(%s:%s)#%s",
                element.fileName,
                element.lineNumber,
                element.methodName)
    }
}